### ISSUE TITLE
Translate “search product…”

### ISSUE THREAD

**Issue by NinjaAtWork** [woocommerce/woocommerce-gutenberg-products-block#7773]
“Search product…”  in input form of woocommerce search block can’t be translate .
Nor the woocommerce translate neither through the woocommerce block plugin translate by loco has no effect
Help how to translate this sentence.

**Comment by nielslange** [comment]
Hello @NinjaAtWork 👋

Thanks for taking time to reach out. I just looked up this issue on a fresh WordPress installation using German and I do see the translation appearing in the Product Search block:

<img width="654" alt="Screenshot 2022-11-29 at 18 10 55" src="https://user-images.githubusercontent.com/3323310/204514225-a5518a97-44d4-4cb8-ade4-34ed3a6916b2.png">

I went ahead, installed the [Loco Translate plugin](https://wordpress.org/plugins/loco-translate/), looked up the German translation for `Search products…`, changed it from `Produkte suchen …` to `Produkte durchsuchen …` and then removed the Product Search block and re-added it again. Doing that resulted in the expected translation:

<img width="1438" alt="Screenshot 2022-11-29 at 18 18 58" src="https://user-images.githubusercontent.com/3323310/204515744-38287259-761e-4b81-b88f-3ed6a8991325.png">
<img width="724" alt="Screenshot 2022-11-29 at 18 19 15" src="https://user-images.githubusercontent.com/3323310/204515803-1cc24713-9f83-48c8-9439-0237ae6b8b63.png">

To troubleshoot the problem you're facing, can you please post the exact steps that you tried? Is it possible that you're using a language where this particular string is not translated yet? In terms of Loco Translate, have you removed and re-added the corresponding block after you updated the translation?

**Comment by NinjaAtWork** [comment]
thanks for reply
if you see German translate without Loco then why i cant see Persian translate?
is there any .po / .mo for Persian which i am missing? it will read the translate from woocommerce itself or from the woocommerce block ?
i used a fresh install like you but it was in English while the site is in persian at the other hand woocommerce is 100% in persian translated
then i have installed the woocommerce block plugin and making new language in loco for both custom/system and translate that sentence and no effect in front-end.

![search2](https://user-images.githubusercontent.com/14922997/204754332-813e1c0a-8faf-4ef1-9868-7d8060fba085.png)
![loco-system](https://user-images.githubusercontent.com/14922997/204764675-ea0e54f6-5664-48de-b3e7-1f3ebc53bd24.png)
![loco-custom](https://user-images.githubusercontent.com/14922997/204764690-3544b559-329f-40b1-b893-2a429c6cbf92.png)

what should i do? here i have host stage and everything can be modified with no problem.

**Comment by nielslange** [comment]
> if you see German translate without Loco then why i cant see Persian translate?

The plugin is translated by volunteers via https://translate.wordpress.org/projects/wp-plugins/woo-gutenberg-products-block/. The translation page for Persian can be found on https://translate.wordpress.org/locale/fa/default/wp-plugins/woo-gutenberg-products-block/. Currently, only 14% of the plugin have been translated. That translation page also shows the following information:

> Translations for the readme are published almost immediately. The initial language pack for the plugin will be generated when 90% of the [Stable (latest release)](https://translate.wordpress.org/projects/wp-plugins/woo-gutenberg-products-block/stable/fa/default/) sub-project strings have been translated (currently 14%).

This explains why some translations, e.g. [German](https://translate.wordpress.org/locale/de/default/wp-plugins/woo-gutenberg-products-block/) (88% translated) or [Spanish](https://translate.wordpress.org/locale/es/default/wp-plugins/woo-gutenberg-products-block/) (100% translated) are visible, while other translations, e.g. [Persian ](https://translate.wordpress.org/locale/fa/default/wp-plugins/woo-gutenberg-products-block/) (14% translated), are not visible yet. 

As mentioned before, translations are done by volunteers. In case you want to contribute with translations, on https://make.wordpress.org/polyglots/handbook/translating/glotpress-translate-wordpress-org/#getting-started you can find information on how to get started. 

> is there any .po / .mo for Persian which i am missing? it will read the translate from woocommerce itself or from the woocommerce block ?

When installing Persian via `/wp-admin/options-general.php`, both `fa_IR.mo` and `fa_IR.po` will be fetched into the language folder of your site, which is located under `/wp-content/languages`. Next, you need to head over to `/wp-admin/update-core.php` to fetch the Persian translations for the installed plugins and themes. Given that WooCommerce Blocks hasn’t been translated >= 90%, you’ll not receive the translations or the *.po or *.mo file for WooCommerce Blocks. However, WooCommerce Blocks has a fallback mechanism for translations. If a certain translation is not available in the WooCommerce Blocks translation file, the plugin will look for the corresponding string in the WooCommerce translation files. If you look up `/wp-content/languages/plugins`, you should see the following files:

- woocommerce-fa_IR.mo
- woocommerce-fa_IR.po

> then I have installed the woocommerce block plugin and making new language in loco for both custom/system and translate that sentence and no effect in front-end.

This might be related to caching. I suggest that you update the translation in Loco Translate, then head over to the page/post where you added the Product Search block, delete that block and save the page/post. Next, [please hard refresh the page/post](https://www.documate.org/resources/what-is-a-hard-refresh-how-to-do-a-hard-refresh-in-any-browser). Now, please add the Product Search block again. You should now be able to see the expected translation within the Product Search block.

I hope this helps to get the Product Search block translated.

**Comment by NinjaAtWork** [comment]
thanks for suggestion. 
i think the:
"fallback mechanism" you mentioned is not working here.
because woocommerce translation is at 100% 

honestly we can't install more plugin specially the one that is bundled with woocommerce  just for one sentence translation.
kindly please polish the "search product..." to be loaded from woocommerce .po/.mo in fallback.
or change it as a placeholder just like the wordpress search block.

**Comment by nielslange** [comment]
> "fallback mechanism" you mentioned is not working here.
because woocommerce translation is at 100%

I just double-checked this case and I do see the Product Search block appearing in Persian using the translation from WooCommerce core:

<img width="1512" alt="Screenshot 2022-12-07 at 13 12 28" src="https://user-images.githubusercontent.com/3323310/206102319-752f88fa-e521-4443-9893-6cdc4c1c0164.png">

These are the steps I've taken to get the result as seen in the screenshot:

1. Head over to `/wp-admin/options-general.php` and install Persian.
2. Head over to `/wp-admin/update-core.php`, click on `بررسی دوباره.` and fetch translations within the section `ترجمه‌ها`.
3. Create a page or post and add the Product Search block which is named `جستجوی محصول` in Persian.

**Comment by NinjaAtWork** [comment]
Hi Hello
everything solved after looking at your screenshot.
the problem was due to admin profile language.
website language was Persian but we used to work in ENG always in dashboard and after reverting to Persian language i got able to see "جستجوی محصول" block which was "Search Block" before.
replaced these Blocks then again set ENG as dashboard language.
so much thanks for help and being aside with me.
👍
